core_lib/src/structure/layerbitmap.cpp

```diff
@@ -57,13 +57,28 @@ Status LayerBitmap::saveKeyFrameFile(KeyFrame* keyframe, QString path)
     QString theFileName = fileName(keyframe);
     QString strFilePath = QDir(path).filePath(theFileName);
 
-    if (needSaveFrame(keyframe, strFilePath) == false)
+    BitmapImage* bitmapImage = static_cast<BitmapImage*>(keyframe);
+
+    Status st = needSaveFrame(keyframe, strFilePath);
+    if (st == Status::OK)
     {
         return Status::SAFE;
     }
 
-    BitmapImage* bitmapImage = static_cast<BitmapImage*>(keyframe);
-    Status st = bitmapImage->writeFile(strFilePath);
+    if (bitmapImage->fileName().isEmpty())
+    {
+        bitmapImage->setFileName(strFilePath);
+    }
+    else if(strFilePath != bitmapImage->fileName())
+    {
+        bitmapImage->setFileName(bitmapImage->fileName());
```

Comment:
Can you please explain a bit about this? why does the bitmap image assign the filename to itself?

Reply:
Ah. I think I understand why. You want to keep the previous file path

Reply:
Yes that's correct, because otherwise we'll load the wrong image path and that was essentially the bug.